src/utils/hooks/useProducts.ts

```diff
@@ -0,0 +1,70 @@
+import { useState, useEffect } from 'react';
+import { API_BASE_URL } from '../constants';
+import { useLatestAPI } from './useLatestAPI';
+
+import { Product, ApiProductsData } from '../types';
+
+interface IProducts {
+  data: Product[];
+  isLoading: boolean;
+}
+
+export function useProducts(): IProducts {
+  const { ref: apiRef, isLoading: isApiMetadataLoading } = useLatestAPI();
+  const [products, setProducts] = useState<IProducts>(() => ({
+    data: [],
+    isLoading: true,
+  }));
+
+  useEffect(() => {
+    if (!apiRef || isApiMetadataLoading) {
+      return () => {};
```

Comment:
I think we can remove this function and just return, that way we avoid an empty function from being executed.